server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/helper/ReflectionHelper.java

```diff
@@ -38,6 +41,39 @@ public ReflectionHelper(Operation operation, EventEmitter eventEmitter) {
         this.method = lookupMethod(operationClass, operation);
     }
 
+    public <T> T invokePrivileged(Object... arguments) {
+        final ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        return invokePrivileged(tccl, arguments);
+    }
+
+    public <T> T invokePrivileged(final ClassLoader classLoader, Object... arguments) {
+
+        try {
+            return (T) AccessController
+                    .doPrivileged(new PrivilegedExceptionAction<Object>() {
+                        @Override
+                        public Object run() {
+                            ClassLoader originalTccl = Thread.currentThread()
+                                    .getContextClassLoader();
+                            Thread.currentThread().setContextClassLoader(classLoader);
+
+                            try {
+                                return invoke(arguments);
+                            } catch (Exception ex) {
+                                // TODO: Handle this better.
+                                throw new RuntimeException(ex);
```

Comment:
You may be able to avoid the catch block entirely and have the run method declare that it throws Exception.  Any exception thrown from here should be wrapped in the PrivilegedActionException.

Reply:
Cool, done !